### PR TITLE
feat: convert schema files to .ts in base component package

### PR DESCRIPTION
**Pull request by eljefe223**
# Description
part 1 of: #1632 

Convert schemas over from `.json` to `.ts` files and export them accordingly.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://microsoft.github.io/fast-dna/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-react-base/src/auto-suggest/examples.data.tsx [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import React from "react";
 import { ComponentFactoryExample } from "@microsoft/fast-development-site-react";
-import schema from "./auto-suggest.schema.json";
-import listboxItemSchema from "../listbox-item/listbox-item.schema.json";
+import { autoSuggestSchema, listboxItemSchema } from "../";
```
Comment (chrisdholt) [inline packages/fast-components-react-base/src/auto-suggest/examples.data.tsx]: This path may have issues resolving in certain environments. Can we update the pathing.

**Comment by chrisdholt** [comment]
These files are currently exported as JSON files - should this be a major version change? Anyone importing or referencing the .json file would break from this with what I can see.

File: packages/fast-components-react-base/src/auto-suggest/examples.data.tsx (hunk 1 above)
Comment (eljefe223) [reply packages/fast-components-react-base/src/auto-suggest/examples.data.tsx]: to "../index.ts" ?

**Comment by eljefe223** [comment]
> These files are currently exported as JSON files - should this be a major version change? Anyone importing or referencing the .json file would break from this with what I can see.

We don't currently export the schema files.

File: packages/fast-components-react-base/src/auto-suggest/examples.data.tsx (hunk 1 above)
Comment (chrisdholt) [reply packages/fast-components-react-base/src/auto-suggest/examples.data.tsx]: yeah, I would path it directly to be safe

**Comment by janechu** [comment]
@eljefe223 I would delete these as a cleanup later, give anyone who might be using these an opportunity to fix to a version where they exist, convert to using the exported javascript and then in another version delete them.

File: packages/fast-components-react-base/src/auto-suggest/index.ts [hunk 2]
```diff
@@ -1,4 +1,6 @@
 import AutoSuggest from "./auto-suggest";
+import autoSuggestSchema from "./auto-suggest.schema";
 
+export { autoSuggestSchema };
```
Comment (chrisdholt) [inline packages/fast-components-react-base/src/auto-suggest/index.ts]: I think we should keep naming intuitive here and elsewhere: `AutoSuggestSchema`

File: build/convert-schemas.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        const commonJS = "module.exports = ".concat(
+            fileContents.substring(fileContents.indexOf("{"))
+        );
+        const tempPath = path.resolve(
+            path.dirname(filePath),
+            `.tmp/${path.basename(filePath)}.js`
+        );
+
+        try {
+            fs.writeFileSync(tempPath, commonJS);
```
Comment (nicholasrice) [inline build/convert-schemas.js]: prior to this we should ensure that the tempPath exists:

```
if (!fs.existsSync(path.dirname(tempPath)) {
  fs.mkdir(path.dirname(tempPath))
}
```

File: packages/fast-components-react-base/src/auto-suggest/index.ts (hunk 2 above)
Comment (eljefe223) [reply packages/fast-components-react-base/src/auto-suggest/index.ts]: Chatted offline

**Pull request merged**